<?xml version="1.0"?>
<launch>

    <arg name="start_pt" default= '0, 0, 0, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/q1.yaml" />

    <arg name="planner_algorithm" default="mpc" />

    <group if="$(eval use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>

    <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
    <rosparam command="load" file="$(find svea_core)/resources/param/obstacles.yaml" />
    <arg name="obstacles_list" default="$(find svea_core)/resources/param/obstacles.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/race_track.yaml" />

    <!-- Occupancy grid parameters -->
    <arg name="og_width" default="1270" />
    <arg name="og_height" default="568" />
    <arg name="og_resolution" default="0.05" />
    <arg name="og_origin_x" default="-30.55" />
    <arg name="og_origin_y" default="-11.4" />

    <node pkg="svea_core" type="team_2_floor2_example.py" name="team_2_floor2_example" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>

        <rosparam command="load" file="$(find svea_core)/resources/param/planner.yaml" />
        <param name="planner_algorithm" value="$(arg planner_algorithm)"/>

        <!-- Occupancy grid parameters -->
        <param name="og_width" value="$(arg og_width)"/>
        <param name="og_height" value="$(arg og_height)"/>
        <param name="og_resolution" value="$(arg og_resolution)"/>
        <param name="og_origin_x" value="$(arg og_origin_x)"/>
        <param name="og_origin_y" value="$(arg og_origin_y)"/>

        <rosparam file="$(arg obstacles_list)" command="load" />

        <remap from="~initial_state" to="/initialpose" />
        <remap from="~goal_state" to="/move_base_simple/goal" />
    </node>

</launch>
